File: CzhERP.Application/Entity/PurInvoice.cs
namespace CzhERP.Application.Entity;

[SugarTable("Pur_Invoice", TableDescription = "采购发票表")]
[SugarIndex("index_{table}_IN", nameof(InvoiceNo), OrderByType.Asc, IsUnique = true)]
public class PurInvoice : EntityBase
{
    [SugarColumn(ColumnDescription = "发票号码", Length = 50, IsNullable = false)]
    public string InvoiceNo { get; set; }

    [SugarColumn(ColumnDescription = "关联订单ID")]
    public long? OrderId { get; set; }

    [SugarColumn(ColumnDescription = "关联入库单ID")]
    public long? InboundId { get; set; }

    [SugarColumn(ColumnDescription = "供应商ID", IsNullable = false)]
    public long SupplierId { get; set; }

    [SugarColumn(ColumnDescription = "供应商名称", Length = 100, IsNullable = false)]
    public string SupplierName { get; set; }

    [SugarColumn(ColumnDescription = "发票类型(1增值税专票/2增值税普票/3电子发票)", IsNullable = false)]
    public int InvoiceType { get; set; }

    [SugarColumn(ColumnDescription = "开票日期", IsNullable = false)]
    public DateTime InvoiceDate { get; set; }

    [SugarColumn(ColumnDescription = "不含税金额", IsNullable = false)]
    public decimal Amount { get; set; }

    [SugarColumn(ColumnDescription = "税率", DefaultValue = "0")]
    public decimal TaxRate { get; set; } = 0;

    [SugarColumn(ColumnDescription = "税额", DefaultValue = "0")]
    public decimal TaxAmount { get; set; } = 0;

    [SugarColumn(ColumnDescription = "价税合计", IsNullable = false)]
    public decimal GrandTotal { get; set; }

    [SugarColumn(ColumnDescription = "状态(0待审核/1已审核/2已核销/3已作废)", DefaultValue = "0")]
    public int Status { get; set; } = 0;

    [SugarColumn(ColumnDescription = "备注", Length = 500)]
    public string? Remark { get; set; }

    [Navigate(NavigateType.OneToOne, nameof(OrderId))]
    public PurOrder? Order { get; set; }

    [Navigate(NavigateType.OneToOne, nameof(InboundId))]
    public PurInbound? Inbound { get; set; }

    [Navigate(NavigateType.OneToOne, nameof(SupplierId))]
    public PurSupplier? Supplier { get; set; }
}

File: Web/src/api/purchase/purInvoice.ts
import {useBaseApi} from '/@/api/base';

// 采购发票表接口服务
export const usePurInvoiceApi = () => {
	const baseApi = useBaseApi("purInvoice");
	return {
		// 分页查询采购发票表
		page: baseApi.page,
		// 查看采购发票表详细
		detail: baseApi.detail,
		// 新增采购发票表
		add: baseApi.add,
		// 更新采购发票表
		update: baseApi.update,
		// 删除采购发票表
		delete: baseApi.delete,
		// 批量删除采购发票表
		batchDelete: baseApi.batchDelete,
		// 导出采购发票表数据
		exportData: baseApi.exportData,
		// 导入采购发票表数据
		importData: baseApi.importData,
		// 下载采购发票表数据导入模板
		downloadTemplate: baseApi.downloadTemplate,
	}
}

// 采购发票表实体
export interface PurInvoice {
	// 主键Id
	id: number;
	// 发票号码
	invoiceNo?: string;
	// 关联订单ID
	orderId: number;
	// 关联入库单ID
	inboundId: number;
	// 供应商ID
	supplierId?: number;
	// 供应商名称
	supplierName?: string;
	// 发票类型(1增值税专票/2增值税普票/3电子发票)
	invoiceType?: number;
	// 开票日期
	invoiceDate?: string;
	// 不含税金额
	amount?: number;
	// 税率
	taxRate?: number;
	// 税额
	taxAmount?: number;
	// 价税合计
	grandTotal?: number;
	// 状态(0待审核/1已审核/2已核销/3已作废)
	status?: number;
	// 备注
	remark: string;
	// 创建时间
	createTime: string;
	// 更新时间
	updateTime: string;
	// 创建者Id
	createUserId: number;
	// 创建者姓名
	createUserName: string;
	// 修改者Id
	updateUserId: number;
	// 修改者姓名
	updateUserName: string;
}